### PR TITLE
ci: fix broken ci components

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:

- **Purpose**: Currently some of the CI jobs won't run, attempt to fix them
- **Approach**: Changes in GitHub CI files and formatting

**Type of change**'

- [x] Bug fix (non-breaking change which fixes an issue)

**Updates**

- Fixes the existing CI build to be compatible with MAUI. See comments for details.

### Test plan :test_tube:

- CI should be all passing

### Author to check :eyeglasses:

- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:

- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: .github/workflows/dotnet-format.yml [hunk 1]
```diff
@@ -17,5 +17,5 @@ jobs:
           
       - name: Format
         working-directory: ./src
-        run: dotnet format --verify-no-changes 
+        run: dotnet format --verify-no-changes --no-restore
```
Comment (miquelbeltran) [inline .github/workflows/dotnet-format.yml:20]: Adding `--no-restore` fixes a runtime issue. The parameter:

> Doesn't execute an implicit restore before formatting. Default is to do implicit restore.

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -12,14 +12,24 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (miquelbeltran) [inline .github/workflows/dotnet.yml:15]: Switching to windows CI so we can test MAUI build for Android, Windows, iOS and macOS

File: .github/workflows/dotnet.yml [hunk 3]
```diff
@@ -12,14 +12,24 @@ on:
...
       uses: actions/setup-dotnet@v4
       with:
         dotnet-version: 8.0.x
+    - name: Setup JDK
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'microsoft'
+        java-version: '17'
+    - name: Setup Android SDK
+      uses: android-actions/setup-android@v3
```
Comment (miquelbeltran) [inline .github/workflows/dotnet.yml:29]: Setup Android build requirements

File: src/Raygun.Blazor.Maui/Raygun.Blazor.Maui.csproj [hunk 4]
```diff
@@ -1,9 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFrameworks>net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>
+        <TargetFrameworks>net8.0-android</TargetFrameworks>
         <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">
-            $(TargetFrameworks);net8.0-windows10.0.19041.0</TargetFrameworks>
+            $(TargetFrameworks);net8.0-ios;net8.0-maccatalyst;net8.0-windows10.0.19041.0</TargetFrameworks>
+        <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('osx'))">
+            $(TargetFrameworks);net8.0-ios;net8.0-maccatalyst</TargetFrameworks>
```
Comment (miquelbeltran) [inline src/Raygun.Blazor.Maui/Raygun.Blazor.Maui.csproj:8]: When the platform is Windows, it builds for Windows, Android, Mac and iOS. When the platform is osx (macos) it builds for Android, iOS and Mac. On the rest of platforms (basically Linux) it only builds for Android.

Same for the Sample project.

File: src/Raygun.Samples.Blazor.Maui/MauiProgram.cs [hunk 5]
```diff
@@ -29,8 +29,8 @@ public static MauiApp CreateMauiApp()
             builder.Services.AddMauiBlazorWebView();
 
 #if DEBUG
-    		builder.Services.AddBlazorWebViewDeveloperTools();
-    		builder.Logging.AddDebug();
+            builder.Services.AddBlazorWebViewDeveloperTools();
+            builder.Logging.AddDebug();
```
Comment (miquelbeltran) [inline src/Raygun.Samples.Blazor.Maui/MauiProgram.cs:33]: Fixes formatting

**Comment by sumitramanga** [comment]
When viewing the Files Changed tab, there are some warnings to address, could you perhaps look into those. Or, if they're an Issue already, I'm happy with them being in a separate branch,

**Comment by miquelbeltran** [comment]
I'll address them in this branch @sumitramanga

**Comment by miquelbeltran** [comment]
@sumitramanga the warnings have been addressed now

**Comment by miquelbeltran** [comment]
Yep, I tried to make it work on Ubuntu but compiling on Linux only works for Android MAUI apps (and only from CLI, not even in Rider). Windows VMs seem the best compromise as it can build for all 4 platforms.

**Pull request merged**